backend/routes/api/users.js

```diff
@@ -93,4 +85,125 @@ router.get('/current', restoreUser, (req, res) => {
+			res.json(user.ownedGames);
+		} else {
+			res.json({ message: 'userId not found' })
+		}	
+	} catch (err){
+		console.log(err) 
+		res.json({ message: `Error fetching user's owned games list` })
+	}
+})
+
+// Add a game to user's owned game list
+router.post('/:userId/ownedGames', async (req, res) => {
+	try {
+		if(req.body.userId !== req.params.userId) {
+			return res.json({ message: 'user ids do not match' })
+		}
+		const user = await User.findById(req.body.userId);
+		const gameId = req.body.gameId;
+
+		if(user.ownedGames.includes(gameId)) {
+			return res.json({ message: 'User already owns this game'})
+		} 
+		user.ownedGames.push(gameId);
+		await user.save();
+
+		return res.json(user);
+	} catch (err) {
+		return res.json({ message: `Error posting to user's owned games list` });
+	}
+})
+
+// Remove a game from a user's owned game list
+router.delete('/:userId/ownedGames', async (req, res) => {
+	try {
+		const user = await User.findById(req.body.userId);
+		const gameId = req.body.gameId;
+
+		const index = user.ownedGames.indexOf(gameId)
+		if(index > -1) {
+			user.ownedGames.splice(index, 1);
+			console.log('Game has been removed');
+		}
+		await user.save();
+		return res.json(user);
+	} catch (err) {
+		console.log(error);
+		return res.json({ message: `Error deleting from user's owned games list`})
+	}
+})
+
+// Add a game to user's wishlist
+router.post('/:userId/wishlistGames', async (req, res) => {
+	try {
+		if(req.body.userId !== req.params.userId) {
+			return res.json({ message: 'user ids do not match' })
+		}
+		const user = await User.findById(req.body.userId);
+		const gameId = req.body.gameId;
+
+		if(user.wishlistGames.includes(gameId)) {
+			return res.json({ message: 'Game already in wishlist'})
+		} 
+		user.wishlistGames.push(gameId);
+		await user.save();
+
+		return res.json(user);
+	} catch (err) {
+		return res.json({ message: `Error posting to user's wishlist` });
+	}
+})
+
+// Remove a game from a user's wishlist
+router.delete('/:userId/wishlistGames', async (req, res) => {
+	try {
+		const user = await User.findById(req.body.userId);
+		const gameId = req.body.gameId;
+
+		const index = user.wishlistGames.indexOf(gameId)
+		if(index > -1) {
+			user.wishlistGames.splice(index, 1);
+			console.log('Game has been removed');
+		}
+		await user.save();
+		return res.json(user);
+	} catch (err) {
+		console.log(error);
+		return res.json({ message: `Error deleting from user's wishlist`})
+	}
+})
+
+// get user by userId
+router.get('/:userId', async (req, res) => {
+	try {
+		const user = await User.findById(req.params.userId)
+		if (user) {
+			res.json(user);
+		} else {
+			res.json({ message: 'userId not found' })
+		}
+	} catch (err) {
+		console.log(err);
+		res.json({ message: 'Error fetching userId'})
+	}
+})
+
+/* get all users */
+router.get('/', async function (req, res, next) {
+	try {
+		const users = await User.find()
+
+		return res.json(users);
+	} catch (err) {
+		res.json([]);
+	}
+});
```

Comment:
We don't need to update this PR but let's put these get user endpoints at the top of the file. The more general routes should be at the top and more specific routes after that.